src/util/encoder.ts

```diff
@@ -22,6 +22,8 @@ const encodeField = (data, key) => {
       return { stringValue: value }
     case '[object Date]':
       return { stringValue: (value as Date).toJSON() }
+    case '[object BigNumber]':
+      return { stringValue: value.toJSON() }
```

Comment:
Let's not start to specify how to encode for every special type.
The lib should check if the function `toJSON` is implemented and call it if it is. Otherwise, maybe simple call `JSON.stringify`?
Let's do this in a new PR